sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java

```diff
@@ -187,7 +193,26 @@ private String getPath(InputComponent component) {
       InputModule module = (InputModule) component;
       return moduleHierarchy.relativePath(module);
     }
-    throw new IllegalStateException("Unkown component: " + component.getClass());
+    throw new IllegalStateException("Unknown component: " + component.getClass());
+  }
+
+  @CheckForNull
+  private String getProjectRelativePath(DefaultInputComponent component) {
+    if (component instanceof InputFile) {
+      DefaultInputFile inputFile = (DefaultInputFile) component;
+      return inputFile.getProjectRelativePath();
+    }
+
+    Path projectBaseDir = moduleHierarchy.root().getBaseDir();
+    if (component instanceof InputDir) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "InputDir"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)